src/dndbeyond.pyj

```diff
@@ -653,6 +657,7 @@ class Monster(CharacterBase):
         self.lookForActions(stat_block, add_dice, inject_descriptions)
         if add_dice:
             self.lookForSpells(stat_block)
+        self._url = window.location.href

```

Comment:
That won't work in encounter pages or from a character sheet's extra creatures. Need to use the link from the stat block's header instead.